website/content/en/docs/best-practices/observability-best-practices.md

```diff
@@ -0,0 +1,255 @@
+---
+title: "Operator Observability Best Practices"
+linkTitle: "Observability Best practices"
+weight: 6
+description: This guide describes the best practices concepts for adding Observability to operators.
+---
+
+## Operator Observability Best Practices
+
+In this document we will outline what operators require in order to meet the "Deep Insights" capability level and provide best practices and examples for creating metrics, recording rules and alerts. It is based on the general guidelines in [Operator Capability Levels](https://sdk.operatorframework.io/docs/overview/operator-capabilities/).
+
+**Note:** For technical documentation of how to add metrics to your operator, please read the [Metrics](https://book.kubebuilder.io/reference/metrics.html) section of the Kubebuilder documentation.
+
+### Deep Insights capability level requirements
+
+1. **Health and Performance metrics** for all of the operator components - Implemented based on the guidelines below.
+2. **Metrics Documentation** - All metrics should have documentation.
+3. **Alerts** for when things are not working as expected for each of the operator's components - Implemented based on the guidelines below.
+4. **Alerts Runbooks** - Each alert MUST include a runbook_url annotation and a runbook.
+5. **Alerts and Metrics Tests** - E2E Testing for metrics and alerts and unit tests for alerts.
+6. **Events** - Custom Resources MUST emit custom events for the operations taking place.
+7. **Metering** - Operator leverages Operator Metering.
+
+   **Note:** Metering isn't mandatory at this point in order to meet the "Deep Insights" capability level requirments.
+
+### Metrics Guidelines
+
+1. Metrics `Help` message should be verbose, since it can be used to create auto generated documentation, Like its done here for example [KubeVirt metrics](https://github.com/kubevirt/kubevirt/blob/main/docs/metrics.md) and generated by [KubeVirt metrics doc generator](https://github.com/kubevirt/kubevirt/blob/main/tools/doc-generator/doc-generator.go).
```

Comment:
```suggestion
1. Metrics `Help` message should be meaningful since it can be used to create auto-generated documentation Like it's done here for example [KubeVirt metrics](https://github.com/kubevirt/kubevirt/blob/main/docs/metrics.md) and generated by [KubeVirt metrics doc generator](https://github.com/kubevirt/kubevirt/blob/main/tools/doc-generator/doc-generator.go).
```



I think you would like to say here that the help should be provided and that it can be used to auto-generate documentation. Therefore, I think the word "verbose" does not fit well.. maybe meaningful? See that we can have a very verbose text that is not meaningful at all. Therefore, an explanation/description does not need to be verbose to be good.

Reply:
About the examples, have we some examples of it in the Prometheus documentation itself? Or is its official good practices and guidelines docs?

Reply:
@camilamacedo86 examples for the help message you mean?

Reply:
@camilamacedo86  The Prometheus guidelines are not that verbose. We added more details.
We can add later examples from the Memchached operator. I'll make sure that Assaf adds a full description to the new metrics.